### PR TITLE
Move addcmul to Aten

### PR DESCRIPTION
**Pull request by VitalyFedyunin**
Move CPU implementation of the `addcmul` operator to Aten ( #22797 )

### before

```python
In [11]: timeit x.addcmul(a, b)                                                                                                                           
1.31 ms ± 18.3 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```

### after

```python
In [9]: timeit x.addcmul(a, b)                                                                                                                  
588 µs ± 22.8 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```

Adding custom code for the case when `value == 1`, doesn't provide significant performance gain.

### CODE REVIEW

File: aten/src/ATen/native/cpu/PointwiseOpsKernel.cpp [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+namespace at {
+namespace native {
+namespace {
+
+static void addcmul_cpu_kernel(TensorIterator& iter, Scalar value) {
+  ScalarType dtype = iter.dtype(0);
+  AT_DISPATCH_ALL_TYPES(dtype, "addcmul_cpu_out", [&] {
+    scalar_t scalar_val = value.to<scalar_t>();
+    at::native::cpu_kernel(
```
Comment (xuhdev) [inline aten/src/ATen/native/cpu/PointwiseOpsKernel.cpp]: nit: no need to write at::native::

File: aten/src/ATen/native/PointwiseOps.cpp [hunk 2]
```diff
@@ -0,0 +1,56 @@
+#include <ATen/native/PointwiseOps.h>
```
Comment (xuhdev) [inline aten/src/ATen/native/PointwiseOps.cpp]: Can we add a comment here? Something like:

> Ternary and higher-order pointwise operations.

File: aten/src/ATen/native/PointwiseOps.cpp [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+Tensor& addcmul_cpu_out(
+    Tensor& result,
+    const Tensor& self,
+    const Tensor& tensor1,
+    const Tensor& tensor2,
+    Scalar value) {
+  assert_no_internal_overlap(result, "addcmul_cpu");
+  checkBackend("bitwise_not", result, self.type().backend());
```
Comment (xuhdev) [inline aten/src/ATen/native/PointwiseOps.cpp]: "bitwise_not" --> "addcmul"?

**Review by ifedan — APPROVED** [review]

**Review by ifedan — CHANGES_REQUESTED** [review]
Do you have tests?

**Comment by VitalyFedyunin** [comment]
> Do you have tests?

Already tested by common_methods_invocations.py

**Review by gchanan — APPROVED** [review]
I'm curious -- wouldn't it only be a few more lines to move the CUDA implementation as well?

**Comment by VitalyFedyunin** [comment]
> I'm curious -- wouldn't it only be a few more lines to move the CUDA implementation as well?

Only to show how to separate migration, and have this PR as example of it.

**Pull request closed without merging**